engine/src/main/java/org/destinationsol/world/generators/PlanetGenerator.java

```diff
@@ -15,25 +15,244 @@
+    }
+
+    /**
+     * This calculates the proper speed for a planet to orbit around the center of the SolarSystem. It is randomized
+     * slightly, so not all planets orbit at the same speed.
+     * @return float representing orbit speed
+     */
+    protected float calculateDefaultPlanetOrbitSpeed() {
+        return SolMath.arcToAngle(PLANET_SPEED, getDistanceFromSolarSystemCenter()) * SolMath.toInt(SolRandom.seededTest(.5f));
+    }
+
+    /**
+     * This calculates the proper speed for a planet to rotate around its axis. It is randomized slightly, so not all
+     * planets rotate at the same speed.
+     * @return float representing rotation speed
+     */
+    protected float calculateDefaultPlanetRotationSpeed() {
+        return SolMath.arcToAngle(PLANET_GROUND_SPEED, getGroundHeight()) * SolMath.toInt(SolRandom.seededTest(.5f));
+    }
+
+    /**
+     * This method modifies the density of low orbit ships that generate on the Planet. The density represents how
+     * frequently the ships spawn
+     * @param densityMultiplier amount by which the current density should be multiplied. Must be at least 0
+     */
+    protected void modifyLowOrbitShipsDensity(float densityMultiplier) {
+        if (densityMultiplier >= 0) {
+            for (ShipConfig shipConfig : getPlanetConfig().lowOrbitEnemies) {
+                shipConfig.density *= densityMultiplier;
+            }
+        } else {
+            logger.error("Ship density multiplier must be at least 0");
+        }
+    }
+
+    /**
+     * This method modifies the density of high orbit ships that generate on the Planet. The density represents how
+     * frequently the ships spawn
+     * @param densityMultiplier amount by which the current density should be multiplied. Must be at least 0
+     */
+    protected void modifyHighOrbitShipsDensity(float densityMultiplier) {
+        if (densityMultiplier >= 0) {
+            for (ShipConfig shipConfig : getPlanetConfig().highOrbitEnemies) {
+                shipConfig.density *= densityMultiplier;
+            }
+        } else {
+            logger.error("Ship density multiplier must be at least 0");
+        }
+    }
+
+    /**
+     * This method modifies the density of ground enemies (turrets) that generate on the Planet. The density represents how
+     * frequently the turrets spawn
+     * @param densityMultiplier amount by which the current density should be multiplied. Must be at least 0
+     */
+    protected void modifyGroundEnemiesDensity(float densityMultiplier) {
+        if (densityMultiplier >= 0) {
+            for (ShipConfig shipConfig : getPlanetConfig().groundEnemies) {
+                shipConfig.density *= densityMultiplier;
+            }
+        } else {
+            logger.error("Ship density multiplier must be at least 0");
+        }
+    }
+
+    /**
+     * This method modifies the percentage of the atmosphere at which low orbit ships start spawning.
```

Comment:
The _percentage_ of the atmosphere? Huh. As in, related to height/distance from the surface of a planet? I guess in theory you could think of an atmosphere as thinning out (thus lower percentage the higher you go), but I wonder if a simply height offset or something would be clearer (I could imagine exotic / magical atmospheres that could be equally thick from surface to top...) . As usual probably a separate refactor.